src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -7,90 +7,124 @@
  
 package frc.robot.subsystems;
 
+import com.ctre.phoenix6.configs.MotionMagicConfigs;
 import com.ctre.phoenix6.configs.MotorOutputConfigs;
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
 import com.ctre.phoenix6.hardware.TalonFX;
 import com.ctre.phoenix6.signals.InvertedValue;
 import com.ctre.phoenix6.signals.NeutralModeValue;
-import edu.wpi.first.math.controller.ArmFeedforward;
-import edu.wpi.first.math.controller.ProfiledPIDController;
-import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import com.nrg948.preferences.RobotPreferences;
+import com.nrg948.preferences.RobotPreferencesLayout;
+import com.nrg948.preferences.RobotPreferencesValue;
+import edu.wpi.first.networktables.GenericEntry;
 import edu.wpi.first.wpilibj.DutyCycleEncoder;
-import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.parameters.ArmParameters;
 
-public class Arm extends SubsystemBase {
+@RobotPreferencesLayout(groupName = "Arm", row = 5, column = 0, width = 1, height = 1)
+public class Arm extends SubsystemBase implements ShuffleboardProducer {
+  @RobotPreferencesValue
+  public static final RobotPreferences.BooleanValue ENABLE_TAB =
+      new RobotPreferences.BooleanValue("Arm", "Enable Tab", false);
 
+  private final String name;
   private final TalonFX motor;
   private final DutyCycleEncoder absoluteEncoder;
-
-  private final TrapezoidProfile profile;
-
-  private final ArmFeedforward feedForward;
-  private final ProfiledPIDController controller;
-
-  private final TrapezoidProfile.State currentAngle = new TrapezoidProfile.State();
-  private final TrapezoidProfile.State goalAngle = new TrapezoidProfile.State();
+  private double currentAngle = 0;
```

Comment:
We should add a member variable for `currentVelocity`, update it in `updateSensorState()` and display it in the `Status` layout of Shuffleboard.